app/controllers/application_controller.rb

```diff
@@ -1,6 +1,41 @@
 class ApplicationController < ActionController::API
 
   private
+  set_current_tenant_through_filter
+  before_action :set_the_current_tenant
+
+  def set_the_current_tenant
+    return unless ENV["ENFORCE_TENANCY"]
+    return unless ident = user_identity
+
+    tenant = Tenant.find_by(:external_tenant => ident)
+    if tenant
+      set_current_tenant(tenant)
+    else
+      render :json => { :errors => "Unauthorized" }, :status => :unauthorized
+    end
+  end
+
+  def user_account_number
```

Comment:
@gtanzillo - It appears this method is no longer in use.  It was also incorrectly calling decode on a nested key. Line 36 decodes correctly.